nixos/modules/services/web-servers/caddy/default.nix

```diff
@@ -41,6 +41,12 @@ let
     in
       "${if pkgs.stdenv.buildPlatform == pkgs.stdenv.hostPlatform then Caddyfile-formatted else Caddyfile}/Caddyfile";
 
+  etcConfigFile = "caddy/Caddyfile";
+
+  configPath = if cfg.enableReload
+    then "/etc/${etcConfigFile}"
+    else configFile;
```

Comment:
```suggestion
    else cfg.configFile;
```

Reply:
If cfg.configFile is specified by users as a json config, and you rename it to `Caddyfile`, will it still work with adapter being null?

Reply:
It doesn’t work indeed, because caddy tries to load the config file in caddyfile format. I’m not sure how I can fix this though. I thought I could do `etcConfigFile = "caddy/${baseNameOf cfg.configFile}";` but this fails (and it would contain the config file hash anyway).

Here are the solutions I could think of:

A. Use a different name for the file in /etc/, eg. `/etc/caddy/caddy_config`. This means we should add a note in the `enableReload` option to mention users will need to set `adapter = "caddyfile"` if they’re using this option and wish to use the caddyfile syntax.
B. Have a dynamic default on `adapter`: `null` when `enableReload = false`, `"caddyfile"` when `enableReload = true`. Not sure it’s doable and/or clean.
C. Change the default value of `adapter` to `"caddyfile"`, and change the file name of `configFile` to something else than `Caddyfile` so we never use caddy’s automatic adapter mechanism.

Option C is very backwards incompatible, option B is probably not feasible, so that leaves only option A, unless you have another idea?

Reply:
The current problem is that:
- when `cfg.adapter` is `null`, `cfg.configFile` can be a json, a Caddyfile specified by a string `"/path/to/CaddyFile"` or a Caddyfile generated by this module at `/nix/store/hash-name/Caddyfile`;
- the config file exposed in `/etc` is named `Caddyfile`;
- when `cfg.adapter` is `null`, `--adapter` is not specified, which cause `caddy` to always use the `caddyfile` adapter and errors on a json.

The solution I propose is as follows:
1. do not name the exposed config file `Caddyfile`
2. use this as the default value of `adapter` option:
```nix
if (builtins.baseNameOf cfg.configFile) == "Caddyfile"
then "caddyfile"
else null
```

Note that this solution is backward compatible so `enableReload` can default to `true`.

Reply:
Cool, I didn’t know about `defaultText` so I was getting an error saying that the default value of an option can’t depend on another option. It works fine when adding `defaultText`.